test/nft-custom.test.js

```diff
@@ -0,0 +1,134 @@
+const {expect} = require("chai");
+
+const caller = "0x5FDd0881Ef284D6fBB2Ed97b01cb13d707f91e42";
+const baseURI = "ipfs://";
+const baseURIUpdated = "https://someipfs.com/mockhash/";
+
+const deploy = async(isFreezeTokenUris = true, overrideBaseURI = null) => {
+  const [owner] = await ethers.getSigners();
+  const NFT = await ethers.getContractFactory("NftCustom");
+
+  const nft = await NFT.deploy("NFTPort", "NFT", owner.address, isFreezeTokenUris, overrideBaseURI !== null ? overrideBaseURI : baseURI);
+  await nft.deployed();
+  return nft;
+}
+
+describe("NftCustom", function () {
+  it("It should deploy the contract, mint a token, and resolve to the right URI", async () => {
+    const nft = await deploy();
+    const URI = "QmWJBNeQAm9Rh4YaW8GFRnSgwa4dN889VKm9poc2DQPBkv";
+    await nft.mintToCaller(caller, 1, URI);
+    expect(await nft.tokenURI(1)).to.equal(baseURI + URI)
```

Comment:
I think in unit tests it make sense to be verbose, so instead of `baseUri + URI` you can have the full thing written, for example: .to.equal(`ipfs://QmWJBNeQAm9Rh4YaW8GFRnSgwa4dN889VKm9poc2DQPBkv`).

But it's fine either way.